services/common/src/main/java/org/finra/gatekeeper/GatekeeperCommonConfig.java

```diff
@@ -129,9 +134,9 @@ public AWSSecurityTokenServiceClient awsSecurityTokenServiceClient() {
     public FilterRegistrationBean userProfileFilterRegistration() {
         FilterRegistrationBean userProfileFilterRegistration = new FilterRegistrationBean();
         if(contentSecurityPolicy != null && !contentSecurityPolicy.isEmpty()) {
-            userProfileFilterRegistration.setFilter(new UserHeaderFilter(new SSOParser(userIdHeader), contentSecurityPolicy));
+            userProfileFilterRegistration.setFilter(new UserHeaderFilter(new SSOParser(userIdHeader, userMembershipsGroupHeader, userEmailHeader, userNameHeader), contentSecurityPolicy));
         } else {
-            userProfileFilterRegistration.setFilter(new UserHeaderFilter(new SSOParser(userIdHeader)));
+            userProfileFilterRegistration.setFilter(new UserHeaderFilter(new SSOParser(userIdHeader, userMembershipsGroupHeader, userEmailHeader, userNameHeader)));
```

Comment:
I feel like I would have a separate SSOParser implementation for when we are using headers for user authorization vs for when we aren't